tests/rptest/tests/e2e_topic_recovery_test.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2021 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.cluster import cluster
+from ducktape.mark import matrix
+from ducktape.cluster.cluster_spec import ClusterSpec
+from rptest.clients.types import TopicSpec
+from rptest.services.redpanda import RedpandaService, SISettings
+from rptest.util import Scale
+from rptest.clients.rpk import RpkTool
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.franz_go_verifiable_services import FranzGoVerifiableProducer, FranzGoVerifiableSeqConsumer
+from rptest.util import wait_until
+import time
+import json
+
+
+class EndToEndTopicRecovery(RedpandaTest):
+    """
+    The test produces data and makes sure that all data
+    is uploaded to S3. After that the test recreates the
+    cluster (or topic) and runs the verifier.
+    """
+
+    topics = (TopicSpec(
+        partition_count=1,
+        replication_factor=3,
```

Comment:
nit: the topic spec by default does `partition_count=1` and `replication_factor=3`